src/github.com/matrix-org/go-neb/services/slackapi/slackapi.go

```diff
@@ -53,6 +56,20 @@ func (s *Service) OnReceiveWebhook(w http.ResponseWriter, req *http.Request, cli
 	w.WriteHeader(200)
 }
 
+// Register joins all configured rooms
+func (s *Service) Register(oldService types.Service, client *matrix.Client) error {
+	for _, mapping := range s.Hooks {
+		if _, err := client.JoinRoom(mapping.RoomID, "", ""); err != nil {
+			log.WithFields(log.Fields{
+				log.ErrorKey: err,
+				"room_id":    mapping.RoomID,
+				"user_id":    client.UserID,
+			}).Error("Failed to join room")
+		}
+	}
+	return nil
```

Comment:
But there is no 1-1 mapping between slackapi service instances and hooks; one service could have many hooks... Given `webhookEndpointURL := baseURL + "services/hooks/" + base64ServiceID`, I think it would make more sense to not have any concept of hook IDs and instead just have a single room ID stored per service. The service ID can then be used as a replacement for this hook ID concept.